File: Application/SkuciSe/app/src/main/java/com/example/skucise/models/Comment.kt
package com.example.skucise.models

import java.io.Serializable

class Comment(
    var id: Long? = null,
    var content: String,
    var rating: Double,
    var datePublished: String,
    var user: User? = null,
    var advert: Advert? = null,
    var userId: Long,
    var advertId: Long,
    var reservationId: Long,
): Serializable {
}

File: Application/Backend/SkuciSeAPI/SkuciSeAPI/Models/Comment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace SkuciSeAPI.Models
{
    public class Comment
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }
        public string Content { get; set; }
        public double Rating { get; set; }
        public DateTime DatePublished { get; set; }

        // Foreign keys
        public long UserId { get; set; }
        public User User { get; set; }

        public long AdvertId { get; set; }
        public Advert Advert { get; set; }

        [ForeignKey("Reservation")]
        public long ReservationId { get; set; }
        public Reservation Reservation { get; set; }
    }
}
